### ISSUE TITLE
Inconsistency between Loss Calculation in Code and Article?

### ISSUE THREAD

**Issue by AlejandroSarrio** [Extrality/AirfRANS#1]
Hi there,

I noticed a discrepancy in the loss calculation in the code compared to what was described in the research paper. I believe this may be a bug, and I wanted to bring it to your attention.

In the section of code you can find here. https://github.com/Extrality/AirfRANS/blob/29f29e4de6627d4c0703869bf18b8f5c439ecd76/train.py#L41-L54 It seems that the condition for `if criterion == 'MSE_weighted'` will never be met. This is because criterion has already been assigned with either `criterion = nn.MSELoss(reduction='none')` or `criterion = nn.SmoothL1Loss(reduction='none')`.

As a result, the model will be backpropagated by the `total_loss`, which is different from what was described in the article. In the MLP full experiment run by me manualy, it seems that the result of the experiment is also based on the total loss, rather than the loss separated by volume loss and surface loss.

**Comment by FlorentExtrality** [comment]
Hi!
You are very right, thank you for your in-depth reading of the code. The experiments and the results given in the paper are done with a classical MSE loss without the separation between the points over the surface and the volume. I did all the experiments again with the bug corrected last weeks to check it and to get the corrected results.
An updated version of the paper with a disclaimer where the proposed loss is defined and a new appendix with the corrected results have been uploaded on [ArXiv](https://arxiv.org/abs/2212.07564) and [HAL](https://hal.science/hal-03806092/).

The repository has also been updated with the bug corrected and some cleaning.

This was an important issue, thank you for pointing it out!!

Best,
Florent
